### ISSUE TITLE
a11y: TextInput with IconButton makes textField unreachable via screenreader

### ISSUE THREAD

**Issue by andywolff** [flutter/flutter#39918]
Internal: b/140829496

Possibly shares a root cause issue #38940 (https://github.com/flutter/flutter/issues/38940#issuecomment-523567392)

If you have a textField with a suffix icon button
```
TextField(
  decoration:InputDecoration(
    suffixIcon: IconButton(...),
  ),
)
```

Then TalkBack is able to focus only on the icon button, and there's no way to activate the text field.

**Comment by justinmc** [comment]
There is an existing issue related to gestures passing through prefix/suffix buttons to the underlying TextField.  I'm not sure how Flutter decides what is reachable by screenreader, maybe it has nothing to do with that, but I'm putting it out there just in case: https://github.com/flutter/flutter/issues/36324

**Comment by jkurtw** [comment]
Internal: b/140829496

**Comment by jkurtw** [comment]
Note: screenreader users cannot complete a critical application flow, at least needs a workaround.

**Comment by goderbauer** [comment]
Sounds good. Also: After https://github.com/flutter/flutter/pull/40390 has been submitted, the hintText will be used as label if no labelText has been supplied. So if you prefer the hintText style, you may use that instead.

**Comment by andywolff** [comment]
So, adding a label does fix the issue before you focus on the field, but the issue is still present if the cursor is within the text field and it is empty. If you screenreader-navigate away while the textfield is empty, there is no way to return to it via screenreader until you get the cursor out of the field (which can be hard to do for screenreader users). If there is text in the field, you can get back to it, but not if it is empty.

That's definitely better, but it means users can still get themselves into a state where they can't interact with the field, so I think this is still a customer blocker for us

**Comment by goderbauer** [comment]
I believe what you describe will be fixed by https://github.com/flutter/flutter/pull/40390, which I just submitted.

**Comment by andywolff** [comment]
Verified, this looks like it was fixed by #40390. Thank you! 
We can remove the customer blocking tag, or close this if there is no remaining work
